### PR TITLE
make team links look like per_user_key links

### PR DESCRIPTION
**Pull request by maxtaco**
- have encryption and signing kids in the per_team_key section
- no more boxes
- reverse sigs: yes!
- checks and tests for those checks

### CODE REVIEW

File: lib/team.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 // Generated by IcedCoffeeScript 108.0.11
```
Comment (oconnor663) [inline lib/team.js]: Weird that GH didn't detect this was a generated file. Any chance some unexpected line crept in?

File: src/subkey.iced [hunk 2]
```diff
@@ -83,7 +83,7 @@ exports.SubkeyBase = class SubkeyBase extends Base
         err = new Error "Sibkey KID mismatch: #{a} != #{b}"
       else
         @reverse_sig_kid = rsk
-    else if @need_reverse_sig()
+    else if @need_reverse_sig json
```
Comment (oconnor663) [inline src/subkey.iced]: was this a missing arg bug before?

File: src/team.iced [hunk 3]
```diff
@@ -4,17 +4,62 @@
...
+      await super opts, defer err
+    cb err
+
+  get_field : () -> "per_team_key"
+  get_new_key_section : () -> @per_team_key
+  set_new_key_section : (m) ->
+    m.generation = @kms.generation
+    m.encryption_kid = @kms.encryption.get_ekid().toString('hex')
+    @per_team_key = m
+  get_new_km : () -> @kms?.signing # use the signing KM
```
Comment (oconnor663) [inline src/team.iced]: Does this mean that `helpers/key.iced` will do different things for the signing pubkey than it does for the encryption one?

File: test/files/team.iced [hunk 4]
```diff
@@ -1,22 +1,52 @@
 {alloc,team} = require '../../'
-{KeyManager} = require('kbpgp').kb
+{EncKeyManager,KeyManager} = require('kbpgp').kb
 {make_esc} = require 'iced-error'
 {new_sig_arg} = require './util'
 
+test_klass = ({T,arg, klass, keys}, cb) ->
+  esc = make_esc cb, "test_klass"
+  delete arg.kms
+  delete arg.team.per_team_key
```
Comment (oconnor663) [inline test/files/team.iced]: Help me follow what this is testing?

File: src/subkey.iced (hunk 2 above)
Comment (maxtaco) [inline src/subkey.iced]: no it just didn't need the arg before this change.

File: src/team.iced (hunk 3 above)
Comment (maxtaco) [inline src/team.iced]: We're never using `helpers/key.iced` in concert with teams.

File: test/files/team.iced (hunk 4 above)
Comment (maxtaco) [inline test/files/team.iced]: It's just resettting state every time.  It's testing team sigs with and without the `per_team_key` section

File: lib/team.js (hunk 1 above)
Comment (maxtaco) [inline lib/team.js]: for me it did detect...

**Pull request merged**